src/server/blockchain/AdminWallet.js

```diff
@@ -537,7 +545,7 @@ export class Wallet {
       }
       currentAddress = address
       log.debug(`sending tx from: ${address} | nonce: ${nonce}`, { uuid, balance, gas, gasPrice })
-      return new Promise((res, rej) => {
+      return await new Promise((res, rej) => {
```

Comment:
this is meaningless
the only reason  to do that is if you add  try{}  block and you  want to catch the  promise exception in this method  and not in  the calling method

Reply:
that's this case exactly - this promise is wrapped onto try (https://github.com/GoodDollar/GoodServer/pull/213/files/495bad0cbb969791add405f08ce770ac06b4b4ff#diff-d1f631eb750beebf5faaaf8f293ba4e0R523) .. catch (https://github.com/GoodDollar/GoodServer/pull/213/files/495bad0cbb969791add405f08ce770ac06b4b4ff#diff-d1f631eb750beebf5faaaf8f293ba4e0R587)

if not await (as it was before fix), only the sync errors during tx.send call will be caught.
The promise will execute out of the main function's scope ('in background') so its rejections will be uncaught b y the try .. catch i've referenced. This will cause unhandled promise rejection, and this was the reason of "This error originated either by throwing inside of an async function without a catch"